pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexType.java

```diff
@@ -236,10 +237,18 @@ public static boolean ignoreDictionaryOverride(boolean optimizeDictionary,
       // Do not create dictionary if index size with dictionary is going to be larger than index size without dictionary
       // This is done to reduce the cost of dictionary for high cardinality columns
       // Off by default and needs optimizeDictionary to be set to true
-      if (fieldSpec.isSingleValueField() && fieldSpec.getDataType().isFixedWidth()) {
-        // if you can safely enable dictionary, you can ignore overrides
-        return canSafelyCreateDictionaryWithinThreshold(cardinality, totalNumberOfEntries,
-            noDictionarySizeRatioThreshold, fieldSpec);
+      if (fieldSpec.isSingleValueField()) {
+        if (fieldSpec.getDataType().isFixedWidth()) {
+          // if you can safely enable dictionary, you can ignore overrides
+          return canSafelyCreateDictionaryWithinThreshold(cardinality, totalNumberOfEntries,
+              noDictionarySizeRatioThreshold, fieldSpec);
+        }
+        // Config not set, default to old behavior of create dictionary for var width cols
+        if (noDictionaryCardinalityRatioThreshold == null) {
+          return true;
+        }
+        // variable width type, so create based simply on cardinality threshold size cannot be calculated easily
+        return noDictionaryCardinalityRatioThreshold * totalNumberOfEntries > cardinality;
```

Comment:
Currently this is not applied to metric when `optimizeDictionaryForMetrics` is true. I think you might also want it for metric as it is more common to have high cardinality?

Reply:
Makes sense! added - we haven't required this but I see it being useful